third_party/nvfuser/csrc/type.h

```diff
@@ -135,7 +135,8 @@ bool PointerOf::operator==(const PointerOf& other) const {
 
 enum class KernelIndexMode { INT32, INT64 };
```

Comment:
Not blocking this PR because this PR is already making the logic much more clear, but I think we should consider completely remote `KernelIndexMode` and just use `PrimDataType` wherever `KernelIndexMode` was used.

Reply:
Generally agreed, and I did consider it as well, but technically speaking, `KernelIndexModel` would allow something more than just always int32 or int64, although I don't have any specific idea.